packages/dev/gui/src/2D/advancedDynamicTexture.ts

```diff
@@ -287,6 +287,15 @@ export class AdvancedDynamicTexture extends DynamicTexture {
     public getChildren(): Array<Container> {
         return [this._rootContainer];
     }
+
+    /**
+     * Will return all controls that are inside this texture
+     * @param directDescendantsOnly defines if true only direct descendants of 'this' will be considered, if false direct and also indirect (children of children, an so on in a recursive manner) descendants of 'this' will be considered
+     * @param predicate defines an optional predicate that will be called on every evaluated child, the predicate must return true for a given child to be part of the result, otherwise it will be ignored
+     * @returns all child controls
+     */
+    public getDescendants<T extends Control>(directDescendantsOnly?: boolean, predicate?: (control: Control) => boolean): T[];

```

Comment:
Relating to my previous comment about getDescendants, I think this is one method that shouldn't be overloaded, as we don't have the guarantee that all descendants will have the same type T.

Reply:
we dont need guarantee that all descendants have the same type T， just need to make sure that the result of geDescendant have same type T.

for example, get all descendant image controls,  such as 
```ts
getDescendant(false, (control) => control instance of Image))
```
through method overloading, we can use type inference to avoid the type conversion, such as 
```ts
const images: Images[] = []; 
images.concat(...getDescendant(false, (control) => control instance of Image)))
```

Reply:
the parameter type of the `predicate`  has still type `Control`, just the result of getDescendants has type T